### PR TITLE
Disallows `while true`

### PR DESCRIPTION
**Pull request by hugoabonizio**
In order to standardize the question and avoid [bikeshedding](https://github.com/crystal-lang/crystal/pull/5146#discussion_r146273623), we should enforce the usage of `loop` statement instead of `while true`.

### CODE REVIEW

**Comment by straight-shoota** [comment]
`while true` and `loop do` are not exactly the same: `loop` uses a block, `while` doesn't. This can make a difference, for example when you have stacked block contexts.

**Comment by hugoabonizio** [comment]
@straight-shoota can you give an example of this difference?

As I see, they are interchangeable 99% of times, including the ocurrencies in this repository.

File: src/ameba/rule/while_true.cr [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  #   do_something
+  #   break if some_condition
+  # end
+  #
+  # And should be replaced by the following:
+  #
+  # loop do
+  #   do_something
+  #   break if some_condition
+  # end
```
Comment (veelenga) [inline src/ameba/rule/while_true.cr]: please use syntax in documentation:

**Comment by veelenga** [comment]
The nice thing about `loop do .. end` is that it is also [will be possible](https://github.com/crystal-lang/crystal/pull/5114) to use inline rescue/ensure/else:

```crystal
loop do
  #
rescue
  #
end
```

instead of:

```crystal
begin
  while true
    #
  end
rescue
  #
end
```

**Comment by hugoabonizio** [comment]
Done :smile:

**Comment by veelenga** [comment]
@hugoabonizio also please add it to [config sample](https://github.com/veelenga/ameba/blob/master/config/ameba.yml). Soon it will be possible to generate it automatically, but for now has to be updated manually (alphabetical order).

@straight-shoota are you able to share an example when `loop do ... end` worse than using `while true` ?

**Comment by straight-shoota** [comment]
I'm sorry, but I don't have a working example. I was just stating that it is not the same and that it *can* make a difference.

Btw. your comparison regarding `rescue` is incorrect, the equivalent with `while` would be:
```crystal
while true
  begin
    #
  rescue
    #
  end
end
```

**Comment by hugoabonizio** [comment]
@straight-shoota I agree that they can be different on specific scenarios, but almost always they are semantically the same (and maybe generate the same code in release mode). Surely this rule can be disabled if that's the case on a project.

@veelenga Added the rule to `ameba.yml`.

**Comment by veelenga** [comment]
> Btw. your comparison regarding rescue is incorrect, the equivalent with while would be:

```crystal
while true
  begin
    #
  rescue
    #
  end
end
```

Didn't know that. Will give it a try 👍

**Pull request merged**